sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/math.scala

```diff
@@ -312,3 +315,90 @@ case class Logarithm(left: Expression, right: Expression)
     """
   }
 }
+
+case class Round(child: Expression, scale: Expression) extends Expression {
+
+  def this(child: Expression) = {
+    this(child, Literal(0))
+  }
+
+  def children: Seq[Expression] = Seq(child, scale)
+
+  def nullable: Boolean = true
```

Comment:
it is not the end of the world to have nullable be more conservative, since it is technically correct to be nullable. however, if there is a way to do a more accurate way to determine nullability, we should do that.